src/hooks/useDisputes.js

```diff
@@ -38,37 +40,99 @@ export default function useDisputes() {
     }
 
     return {
-      disputes: disputes.map((dispute, i) => ({
-        ...dispute,
-        ...processDisputeData(dispute),
-        ...disputesPhases[i],
-      })),
+      disputes: disputes.map((dispute, i) => {
+        const [disputeDescription] = getDisputeInfoFromMetadata(
+          dispute.metadata
+        )
+        return {
+          ...dispute,
+          description: disputeDescription,
+          ...disputesPhases[i],
+        }
+      }),
     }
   }, [disputesPhases, disputes, disputesPhasesKey, error]) // eslint-disable-line react-hooks/exhaustive-deps
 }
 
 export function useDispute(disputeId) {
+  const [disputeProcessed, setDisputeProcessed] = useState()
   const courtConfig = useCourtConfig()
   const now = useNow() // TODO: use court clock
   const { dispute, fetching } = useSingleDisputeSubscription(disputeId)
+  useEffect(() => {
+    const fetchDataFromIpfs = async () => {
+      if (dispute) {
+        const [
+          disputeDescription,
+          disputeMetadata,
+        ] = getDisputeInfoFromMetadata(dispute.metadata)
+
+        if (!disputeMetadata) {
+          return setDisputeProcessed({ ...dispute, error: true })
+        }
+        const ipfsPath = disputeMetadata.replace(/^ipfs:/, '')
+
+        if (isIPFS.cidPath(ipfsPath)) {
+          const { data, error } = await ipfsGet(ipfsPath)
+          if (error) {
+            return setDisputeProcessed({ ...dispute, error: true })
+          }
+          try {
+            const parsedDisputeData = JSON.parse(data)
+            const agreementText = parsedDisputeData.agreementText.replace(
```

Comment:
Ahh, we don't have to do this manually; we should be able to use:

```
const agreementUrl = resolvePathname(
  agreementText,
  `${IPFS_ENDPOINT}/${ipfsPath}/`
)
```

**Note** the end slash at the end :).

------

But we should bail on the agreement text if it's not available (right now this would just throw)

Reply:
Could you explain what do you mean by " dont have to do this manually?" 
which would be the difference with ```const agreementUrl = resolvePathname(
            agreementText,
            `${IPFS_ENDPOINT}/${ipfsPath}`
          )```
besides the `/` at the end?

Reply:
Having the `/` at the end makes `resolvePathname()` behave differently.

```
resolvePathname('a', 'b') // = 'a'
resolvePathname('a', 'b/') // = 'b/a'
resolvePathname('./a', 'b/') // = 'b/a'
resolvePathname('../a', 'b/c') // 'a'
resolvePathname('../a', 'b/c/') // 'b/a'
```

So we don't need do the manual `.replace()` here, because `resolvePathname()` will take care of it for us.